lib/lintPostcssResult.mjs

```diff
@@ -137,5 +138,15 @@ function isFixCompatible({ stylelint: { disabledRanges } }, name) {
  * @returns {void}
  */
 function warnDeprecatedRule(result, ruleName) {
-	result.warn(`The "${ruleName}" rule is deprecated.`, { stylelintType: 'deprecation' });
+	const message = `The "${ruleName}" rule is deprecated.`;
+
+	emitDeprecationWarning(
+		message,
+		'RULE',
+		`Please be aware that the "${ruleName}" rule will soon be either removed or renamed.`,
+	);
+
+	// @ts-expect-error
```

Comment:
[suggest] Let's add a detailed error to help understand the reason for suppressing the error:

```suggestion
	// @ts-expect-error -- TS2339: Property 'noDeprecation' does not exist on type 'Process'.
```

Reply:
Actually the comment under this one shows that it's wrong.
It does exist and it's just missing from the d.ts.
This message would be misleading.

Reply:
I don't think the message would be misleading because `tsc` outputs the error message exactly. We would be rather a bit confused if `@ts-expect-error` has no reason (e.g. "What error does this `@ts-expect-error` suppress?").

Reply:
So we are adding as the reason the exact opposite of why it's being added?
I guess, I could add it, sure.

Reply:
If adding "why" more exactly, we should include why `@types/note` doesn't have `process.noDeprecation`. However, the link to the Node.js doc is enough for me.